src/deprecate.ts

```diff
@@ -14,13 +12,13 @@ var didWarnForDeprecation = {};
  * @return a wrapped function
  */
 function deprecate(fn, key, message) {
-    return function() {
+    return function(...args) {
         if (!didWarnForDeprecation[key]) {
             didWarnForDeprecation[key] = true;
             console.warn(message);
         }
-        fn.apply(this, arguments);
+        fn.apply(this, args);
```

Comment:
What is `this` expected to be in here? I would just replace this line with `fn(...args);` and let the call function have `this` of its own.